src/applications/check-in/hooks/useGetUpcomingAppointmentsData.jsx

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useLayoutEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector, batch } from 'react-redux';
+import { api } from '../api';
+import { makeSelectCurrentContext } from '../selectors';
+
+import { recievedUpcomingAppointments } from '../actions/universal';
+
+const useGetUpcomingAppointmentsData = () => {
+  const [isComplete, setIsComplete] = useState(false);
+  const [
+    upcomingAppointmentsDataError,
+    setUpcomingAppointmentsDataError,
+  ] = useState(false);
+
+  const selectCurrentContext = useMemo(makeSelectCurrentContext, []);
+  const { token } = useSelector(selectCurrentContext);
+
+  const dispatch = useDispatch();
+
+  const setUpcomingAppointmentsData = useCallback(
+    payload => {
+      batch(() => {
+        const { appointments } = payload;
+        dispatch(recievedUpcomingAppointments(appointments, token));
+      });
+    },
+    [dispatch, token],
+  );
+
+  useLayoutEffect(
+    () => {
+      if (token && !isComplete) {
```

Comment:
I think you should be checking if you already have the data before requesting it. Otherwise the page will make a call every time someone goes to it. For example if you click details and then back to list. You could borrow the needsRefresh concept from the other hook or figure out something else. This can also be spun out into another ticket since the focus of this ticket was the display side.

Reply:
I created a ticket https://github.com/department-of-veterans-affairs/va.gov-team/issues/66619